### PR TITLE
UUID, Set, Dictionary, RealmAny Data Types -- Android SDK

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
## Pull Request Info

Staging PR for all Android SDK "New Data Types" once they're fully reviewed.

### Staged Changes (Requires MongoDB Corp SSO)

- [Dictionary](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/android-sdk-uuid-dict-set-any/sdk/android/data-types/realmdictionary/)
- [Set](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/android-sdk-uuid-dict-set-any/sdk/android/data-types/realmset/)
- [Mixed](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/android-sdk-uuid-dict-set-any/sdk/android/data-types/realmany/)
- [Field Types](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/android-sdk-uuid-dict-set-any/sdk/android/data-types/field-types/)
- [Object Models and Schemas -- just the type lists](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/android-sdk-uuid-dict-set-any/sdk/android/fundamentals/object-models-and-schemas/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/android/local/app/src/main/java/com/mongodb/realm/examples/model/java/FrogAny.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// }
+import com.mongodb.realm.examples.model.kotlin.Person;
+
+import io.realm.RealmAny;
+import io.realm.RealmObject;
+
+public class FrogAny extends RealmObject {
+    String name;
+    RealmAny bestFriend;
+    // realm-required empty constructor
```
Comment (mongomoe) [inline examples/android/local/app/src/main/java/com/mongodb/realm/examples/model/java/FrogAny.java:16]: is "    // realm-required empty constructor" meant to be in the generated file as well?

Comment (nathan-contino-mongo) [reply examples/android/local/app/src/main/java/com/mongodb/realm/examples/model/java/FrogAny.java:16]: yes, the whole point is to show that Realm Objects must implement a constructor that accepts zero arguments.

File: source/sdk/android/data-types/field-types.txt [hunk 2]
```diff
@@ -47,3 +50,15 @@ Since {+client-database+} operates on fields as a whole, it's not possible
...
+
+Object IDs and UUIDs
+--------------------
+
+.. note:: UUID type -- *new to version 10.6.0*
+
+``ObjectId`` and ``UUID`` (Universal Unique Identifier) both provide
+unique values that can be used as identifiers for objects.
+``ObjectId`` is a :manual:`MongoDB-specific </reference/method/ObjectId/>`
+12-byte unique value. ``UUID`` is a :rfc:`standardized <4122>` 16-byte
```
Comment (mongomoe) [inline source/sdk/android/data-types/field-types.txt:62]: broken link: "standardized"

Comment (nathan-contino-mongo) [reply source/sdk/android/data-types/field-types.txt:62]:
```suggestion
12-byte unique value. ``UUID`` is a :rfc:`standardized <rfc4122>` 16-byte
```

File: examples/android/local/app/src/main/java/com/mongodb/realm/examples/model/java/FrogAny.java (hunk 1 above)
Comment (mongomoe) [reply examples/android/local/app/src/main/java/com/mongodb/realm/examples/model/java/FrogAny.java:16]: gotcha 👍

**Comment by mongomoe** [comment]
I think this might not be part of this PR but in object models & schemas there is the following typo:
Screen Shot 2021-05-17 at 1 28 23 PM

duplicate 'a': "... cannot define a a primary key"

**Review by mongomoe — APPROVED** [review]
LGTM 👍  (there's that one typo, but it seems like its not a part of this PR itself so up to you on how you want to handle it (whether that's creating a separate PR/ticket, or fixing the typo a part of this PR or something else)

**Pull request merged**